src/tcgwars/logic/impl/gen4/DiamondPearl.groovy

```diff
@@ -521,19 +616,23 @@ public enum DiamondPearl implements LogicCardInfo {
           weakness L, PLUS30
           resistance F, MINUS20
           move "Accelerative Dive", {
-            text "30 damage. until the end of your next turn."
+            text "30 damage. Flip a coin. If tails, this attack does nothing. If heads, prevent all damage done to Staraptor by attacks (both yours and your opponent’s) until the end of your next turn."
             energyCost C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              flip {
+                //TODO: preventAllDamageNextTurn() won't cut it here sadly.
```

Comment:
What if you reapply it on the `BEGIN_TURN` effect for your opponent as a one off delayed effect?